token/token_test.go

```diff
@@ -36,7 +37,49 @@ func TestGenerateToken(t *testing.T) {
 }
 
 func TestExtractWithInvalidToken(t *testing.T) {
-	t.Skip("Not implemented, verify token.Valid")
+	// This tests generates invalid Token
```

Comment:
Could you split this up into separate Test functions per 'Extract'  verification.